gapic/schema/wrappers.py

```diff
@@ -625,8 +625,22 @@ def client_output(self):
     def field_headers(self) -> Sequence[str]:
         """Return the field headers defined for this method."""
         http = self.options.Extensions[annotations_pb2.http]
+
+        pattern = re.compile(r'\{([a-z][\w\d_.]+)=')
+
         if http.get:
-            return tuple(re.findall(r'\{([a-z][\w\d_.]+)=', http.get))
+            return tuple(pattern.findall(http.get))
+        elif http.put:
+            return tuple(pattern.findall(http.put))
+        elif http.post:
+            return tuple(pattern.findall(http.post))
+        elif http.delete:
+            return tuple(pattern.findall(http.delete))
+        elif http.patch:
+            return tuple(pattern.findall(http.patch))
+        elif http.custom.path:
+            return tuple(pattern.findall(http.custom.path))
```

Comment:
This screams "a pattern we can generalize."
How about
```python
potential_verbs = [
    http.get,
    http.put,
    http.post,
    http.delete,
    http.patch,
    http.custom.path,
]
return next((tuple(pattern.findall(verb)) for verb in potential verbs if verb), ())
```
?

Reply:
Wasn't sure how to wrap this up nicely, thank you!

Reply:
I'm the champion of abusing `next` :)